crates/editor/src/editor.rs

```diff
@@ -10386,6 +10386,22 @@ impl Editor {
         cx.notify();
     }
 
+    pub fn working_directory(&self, cx: &WindowContext) -> Option<PathBuf> {
+        if let Some(buffer) = self.buffer().read(cx).as_singleton() {
+            if let Some(file) = buffer.read(cx).file().and_then(|f| f.as_local()) {
+                if let Some(dir) = file.abs_path(cx).parent() {
+                    return Some(dir.to_owned());
+                }
+            }
+
+            if let Some(project_path) = buffer.read(cx).project_path(cx) {
+                return Some(project_path.path.to_path_buf());
```

Comment:
That is supposed to be a file path, IIRC, so sounds like we have to call `parent` here?

Also, you could've used `?` more instead of all `if let Some` constructs.